src/test/java/teammates/test/MockLogsProcessor.java

```diff
@@ -105,4 +117,129 @@ public void createFeedbackSessionLog(String courseId, String email, String fsNam
         return feedbackSessionLogs;
     }
 
+    private Predicate<MockGeneralLogEntry> getPredicate(String severity, String minSeverity, Instant startTime,
```

Comment:
I hate to break it, but this method makes the entire point of mocking this class useless.
This is because now the different test cases are testing _this method_ instead. The purpose of testing is to check whether production code works + mocking areas that are not of interest during testing. Most mock methods are lightweight (few lines of code) for that reason. This method, however, is as complicated and error-prone as the method to be mocked.

Reply:
Understood. It is removed and the test cases test the action class solely.